migration/migration_blackbox_test.go

```diff
@@ -0,0 +1,149 @@
+package migration_test
+
+import (
+	"database/sql"
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/fabric8-services/fabric8-common/gormsupport"
+	"github.com/fabric8-services/fabric8-common/migration"
+	"github.com/fabric8-services/fabric8-common/resource"
+	"github.com/jinzhu/gorm"
+	_ "github.com/lib/pq"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	dbName = "test"
+)
+
+var host, port string
+
+func setupTest(t *testing.T) {
+	host = os.Getenv("F8_POSTGRES_HOST")
+	require.NotEmpty(t, host, "F8_POSTGRES_HOST is not set")
+	port = os.Getenv("F8_POSTGRES_PORT")
+	require.NotEmpty(t, port, "F8_POSTGRES_PORT is not set")
+
+	dbConfig := fmt.Sprintf("host=%s port=%s user=postgres password=mysecretpassword sslmode=disable connect_timeout=5", host, port)
+
+	db, err := sql.Open("postgres", dbConfig)
+	require.NoError(t, err, "cannot connect to database: %s", dbName)
+	defer db.Close()
+
+	_, err = db.Exec("DROP DATABASE " + dbName)
+	if err != nil && !gormsupport.IsInvalidCatalogName(err) {
+		require.NoError(t, err, "failed to drop database '%s'", dbName)
+	}
+
+	_, err = db.Exec("CREATE DATABASE " + dbName)
+	require.NoError(t, err, "failed to create database '%s'", dbName)
+}
+
+func TestMigrate(t *testing.T) {
+	resource.Require(t, resource.Database)
+
+	setupTest(t)
+
+	dbConfig := fmt.Sprintf("host=%s port=%s user=postgres password=mysecretpassword dbname=%s sslmode=disable connect_timeout=5",
+		host, port, dbName)
+
+	sqlDB, err := sql.Open("postgres", dbConfig)
+	require.NoError(t, err, "cannot connect to DB '%s'", dbName)
+	defer sqlDB.Close()
+
+	gormDB, err := gorm.Open("postgres", dbConfig)
+	require.NoError(t, err, "cannot connect to DB '%s'", dbName)
+	defer gormDB.Close()
+
+	dialect := gormDB.Dialect()
+	dialect.SetDB(sqlDB)
+
+	err = migrate(gormDB.DB(), dbName)
+
+	assert.Nil(t, err)
+	checkMigrate(t, gormDB, dialect)
```

Comment:
I guess the content of this function could be moved here since it's only called once.

Reply:
it's about modularity and not about re-usability ;-)